### PR TITLE
Add security policy

### PR DESCRIPTION
**Pull request by lagru**
## Description

Inspired by [Redis' security policy](https://github.com/redis/redis/blob/d27c7413a95a0a271b94376f3ec64dae06326b29/SECURITY.md) which is also covered by the BSD-3-Clause license.

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

Summarize the introduced changes in the code block below in one or a few sentences. The
summary will be included in the next release notes automatically:

```release-note
...
```

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+We may apply a responsible disclosure process to reported or otherwise
+discovered vulnerabilities. We will usually do that for a critical
+vulnerability, and only if we have a good reason to believe information about
+it is not yet public.
+
+--
+
+_We are thankful to
+[Redis, from whose SECURITY.md](https://github.com/redis/redis/blob/d27c7413a95a0a271b94376f3ec64dae06326b29/SECURITY.md)
+we drew content and inspiration._
```
Comment (lagru) [inline SECURITY.md:33]: Not sure if this is enough recognition or if we should also add a copyright to our license file. I basically copied what we do in our CoC.

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+scikit-image core team at: <br>
+skimage-core@discuss.scientific-python.org
+
+We will evaluate your report and if necessary issue a fix and an advisory. If
+the issue was previously undisclosed, we'll also mention your name in the
+release notes.
+
+## Responsible Disclosure
+
+We may apply a responsible disclosure process to reported or otherwise
```
Comment (jarrodmillman) [inline SECURITY.md:24]: What does this mean? Looking on the internet, it seems like that means we would provide companies using our software early warnings before we announce the issue publicly.

**Review by jarrodmillman — COMMENTED** [review]
I am not sure we should make promises like this. I am not sure, but it seems like the project we are getting this text from is supported by a company with over 500 employees:
https://www.crunchbase.com/organization/redis

We are a mostly volunteer project without major companies as are our paying clients.

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+exceptions may be made.
+
+## Reporting a Vulnerability
+
+If you believe you've discovered a serious vulnerability, please contact the
+scikit-image core team at: <br>
+skimage-core@discuss.scientific-python.org
+
+We will evaluate your report and if necessary issue a fix and an advisory. If
+the issue was previously undisclosed, we'll also mention your name in the
```
[the remark this answers is not in the document]
Comment (lagru) [reply SECURITY.md:19]: Happy to take it out.

**Comment by lagru** [comment]
Replying to you both generally: I am totally onboard with keeping commitment and work down. The initial draft was just to start somewhere. Let me know your thoughts on the new version.

Personally, I still think there is merit in having a non-public channel for the community to reach out. If we don't want that, we should probably disable our [advisory board](https://github.com/scikit-image/scikit-image/security/advisories) as well.

Marking this as draft for now and will close if there's no consensus. :)

**Comment by stefanv** [comment]
If there is a very clear reason for us to have / manage security issues, I'm open to it, but so far I haven't seen any compelling evidence. E.g., if this is something that is required of us to get funding, or if it can help address specific issues that we've seen before, that would be helpful context.

I'm fine with closing the advisory board, since it is empty.

Alternatively, we can simply set our security policy to point to Tidelift, as NumPy does.

**Comment by jarrodmillman** [comment]
I didn't know that private reporting mechanism existed. I couldn't find any mention of it in our documentation and I am not sure how we would know if something was reported or who would get a notification. So, I disabled it here: https://github.com/scikit-image/scikit-image/settings/security_analysis

We can easily reenable it, but we should make sure someone knows how it works and document it in our docs.

**Comment by lagru** [comment]
I'm closing this for now as not needed / wanted. :) If we want to add the means to reach out to us via a private channel, we can discuss and implement that in another PR.

**Pull request closed without merging**